src/msgpack_asgi/_middleware.py

```diff
@@ -1,25 +1,40 @@
 import json
+from typing import Any, Callable
 
 import msgpack
 from starlette.datastructures import Headers, MutableHeaders
 from starlette.types import ASGIApp, Message, Receive, Scope, Send
 
 
 class MessagePackMiddleware:
-    def __init__(self, app: ASGIApp) -> None:
+    def __init__(
+        self,
+        app: ASGIApp,
+        packb: Callable[[Any], bytes] = msgpack.packb,
+        unpackb: Callable[[bytes], Any] = msgpack.unpackb,
+    ) -> None:
         self.app = app
+        self.packb = packb
+        self.unpackb = unpackb
 
     async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
         if scope["type"] == "http":
-            responder = _MessagePackResponder(self.app)
+            responder = _MessagePackResponder(self.app, self.packb, self.unpackb)
             await responder(scope, receive, send)
             return
         await self.app(scope, receive, send)
 
 
 class _MessagePackResponder:
-    def __init__(self, app: ASGIApp) -> None:
+    def __init__(
+        self,
+        app: ASGIApp,
```

Comment:
(Nit) We’d probably also want to make these keyword-only, for consistency

```suggestion
        app: ASGIApp,
        *,
```